frontend/src/components/MapComponent.jsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useEffect, useRef } from "react";
+import "ol/ol.css";
+import Map from "ol/Map";
+import View from "ol/View";
+import TileLayer from "ol/layer/Tile";
+import VectorLayer from "ol/layer/Vector";
+import OSM from "ol/source/OSM";
+import VectorSource from "ol/source/Vector";
+import { useGeographic } from "ol/proj";
+import GeoJSON from "ol/format/GeoJSON.js";
+import { Box, Button, MenuItem, Select, Typography } from "@mui/material";
+import PopoverComponent from "./PopOverComponent";
+import DrawComponent from "./DrawComponent";
+import SearchComponent from "./SearchComponent";
+
+const MapComponent = () => {
+  const [map, setMap] = useState(null);
+  const [vectorLayer, setVectorLayer] = useState(null);
+  const [openModal, setOpenModal] = useState(false);
+  const [selectedPlaceProperties, setSelectedPlaceProperties] = useState(null);
+  const [drawing, setDrawing] = useState(false);
+  const anchorElRef = useRef(null);
+
+  useEffect(() => {
+    useGeographic();
+    const mapInstance = new Map({
+      target: "map",
```

Comment:
Take into account employing the _useRef_ hook to interact with your map container Box. Create a _containerRef_ and link it to your Box using the ref property. Consequently, set the _target_ option's value as _containerRef.current_.